include/swift/Serialization/Validation.h

```diff
@@ -140,6 +140,7 @@ class ExtendedValidationInfo {
     unsigned IsAllowModuleWithCompilerErrorsEnabled : 1;
     unsigned IsConcurrencyChecked : 1;
     unsigned HasCxxInteroperability : 1;
+    unsigned OnlyHasExportableDecls: 1;
```

Comment:
This memory is uninitialized. You need to zero it out in ModuleDecl::ModuleDecl() next to the rest of the flags.